### PR TITLE
Fix Database Timeout config

### PR DESCRIPTION
**Pull request by jan-janssen**
Convert from string to int

### CODE REVIEW

**Comment by jan-janssen** [comment]
I use a scriptjob which submits multiple calculation using `srun` within the queue. If I do not set the timeout I get the following error:
```

---------------------------------------------------------------------------

RuntimeError                              Traceback (most recent call last)

/tmp/ipykernel_40204/4210708647.py in <module>

      1 job_lst = []

      2 for i in range(8):

----> 3     job = pr.create.job.Lammps("lmp_" + str(i))

      4     job.structure = structure

      5     job.calc_md(temperature=500.0, n_print=1)

 

~/.conda/envs/myenv/lib/python3.7/site-packages/pyiron_base/job/jobtype.py in wrapper(job_name, delete_existing_job)

    152                     job_class_dict=self._job_class_dict

    153                 )

--> 154                 return job.create(job_name=job_name, delete_existing_job=delete_existing_job)

    155             return wrapper

    156         else:

 

~/.conda/envs/myenv/lib/python3.7/site-packages/pyiron_base/job/jobtype.py in create(self, job_name, delete_existing_job)

    183             job_name=job_name,

    184             job_class_dict=self._job_class_dict,

--> 185             delete_existing_job=delete_existing_job

    186         )

    187

 

~/.conda/envs/myenv/lib/python3.7/site-packages/pyiron_base/job/jobtype.py in __new__(cls, class_name, project, job_name, job_class_dict, delete_existing_job, delete_aborted_job)

     63         else:

     64             raise TypeError()

---> 65         job = job_class(project, job_name)

     66         if job.job_id is not None and not os.path.exists(job.project_hdf5.file_name):

     67             job.logger.warning(

 

~/.conda/envs/myenv/lib/python3.7/site-packages/pyiron_atomistics/lammps/lammps.py in __init__(self, project, job_name)

     33

     34     def __init__(self, project, job_name):

---> 35         super(Lammps, self).__init__(project, job_name)

     36         self.__name__ = "Lammps"

     37         self._executable_activate(enforce=True)

 

~/.conda/envs/myenv/lib/python3.7/site-packages/pyiron_atomistics/lammps/interactive.py in __init__(self, project, job_name)

     37 class LammpsInteractive(LammpsBase, GenericInteractive):

     38     def __init__(self, project, job_name):

---> 39         super(LammpsInteractive, self).__init__(project, job_name)

     40         self._check_opened = False

     41         self._interactive_run_command = None

 

~/.conda/envs/myenv/lib/python3.7/site-packages/pyiron_atomistics/lammps/base.py in __init__(self, project, job_name)

     54

     55     def __init__(self, project, job_name):

---> 56         super(LammpsBase, self).__init__(project, job_name)

     57         self.input = Input()

     58         self._cutoff_radius = None

 

~/.conda/envs/myenv/lib/python3.7/site-packages/pyiron_atomistics/atomistics/job/interactive.py in __init__(self, project, job_name)

     26 class GenericInteractive(AtomisticGenericJob, InteractiveBase):

     27     def __init__(self, project, job_name):

---> 28         super(GenericInteractive, self).__init__(project, job_name)

     29         self.output = GenericInteractiveOutput(job=self)

     30         self._structure_previous = None

 

~/.conda/envs/myenv/lib/python3.7/site-packages/pyiron_atomistics/atomistics/job/atomistic.py in __init__(self, project, job_name)

    122

    123     def __init__(self, project, job_name):

--> 124         super(AtomisticGenericJob, self).__init__(project, job_name)

    125         self.__version__ = "0.1"

    126         self._structure = None

 

~/.conda/envs/myenv/lib/python3.7/site-packages/pyiron_base/job/interactive.py in __init__(self, project, job_name)

    114

    115     def __init__(self, project, job_name):

--> 116         super(InteractiveBase, self).__init__(project, job_name)

    117         self._interactive_library = None

    118         self._interactive_write_input_files = False

 

~/.conda/envs/myenv/lib/python3.7/site-packages/pyiron_base/job/generic.py in __init__(self, project, job_name)

    164         self._executable = None

    165         self._status = JobStatus(db=project.db, job_id=self.job_id)

--> 166         self.refresh_job_status()

    167         self._restart_file_list = list()

    168         self._restart_file_dict = dict()

 

~/.conda/envs/myenv/lib/python3.7/site-packages/pyiron_base/job/generic.py in refresh_job_status(self)

    425         Refresh job status by updating the job status with the status from the database if a job ID is available.

    426         """

--> 427         if self.job_id:

    428             self._status = JobStatus(

    429                 initial_status=self.project.db.get_job_status(self.job_id),

 

~/.conda/envs/myenv/lib/python3.7/site-packages/pyiron_base/job/core.py in job_id(self)

    186         """

    187         if self._job_id is None:

--> 188             self._job_id = self.get_job_id()

    189         return self._job_id

    190

 

~/.conda/envs/myenv/lib/python3.7/site-packages/pyiron_base/job/core.py in get_job_id(self, job_specifier)

    606             }

    607             response = self.project.db.get_items_dict(

--> 608                 where_dict, return_all_columns=False

    609             )

    610             if len(response) > 0:

 

~/.conda/envs/myenv/lib/python3.7/site-packages/pyiron_base/database/generic.py in get_items_dict(self, item_dict, return_all_columns)

    773

    774             result = self.conn.execute(query)

--> 775         row = result.fetchall()

    776         if not self._keep_connection:

    777             self.conn.close()

 

~/.conda/envs/myenv/lib/python3.7/site-packages/sqlalchemy/engine/result.py in fetchall(self)

    990         """A synonym for the :meth:`_engine.Result.all` method."""

    991

--> 992         return self._allrows()

    993

    994     def fetchone(self):

 

~/.conda/envs/myenv/lib/python3.7/site-packages/sqlalchemy/engine/result.py in _allrows(self)

    400         rows = self._fetchall_impl()

    401         if make_row:

--> 402             made_rows = [make_row(row) for row in rows]

    403         else:

    404             made_rows = rows

 

~/.conda/envs/myenv/lib/python3.7/site-packages/sqlalchemy/engine/result.py in <listcomp>(.0)

    400         rows = self._fetchall_impl()

    401         if make_row:

--> 402             made_rows = [make_row(row) for row in rows]

    403         else:

    404             made_rows = rows

 

RuntimeError: number of values in row (16) differ from number of column processors (1)

RuntimeError: number of values in row (16) differ from number of column processors (1)


```

**Comment by pmrv** [comment]
The conversion is definitely a bug on my part, but I don't understand the connection to the error message, since it seems to come from fairly deep within sqlalchemy.  That happens only if the timeout is not specified in the config file?  If so what timeout do you specify?

**Comment by jan-janssen** [comment]
Now I set the time out to 0 then it works. 

You should be able to reproduce it, I create a script job which I send to the queue and then in the script job I have a for loop creating a couple of LAMMPS calculation and each LAMMPS calculation is running in non-modal mode. The LAMMPS executables use srun rather than mpirun to distribute the jobs to the different nodes within the allocation of the scriptjob.

**Comment by pmrv** [comment]
I suppose there's two separate issues, though.  I'll merge this and move the error to an issue.

File: pyiron_base/settings/generic.py [hunk 1]
```diff
@@ -292,12 +292,12 @@ def _config_parse_file(self, config_file):
                 self._configuration["sql_view_user_key"] = parser.get(
                     section, "VIEWERPASSWD"
                 )
-            self._configuration["connection_timeout"] = parser.get(section, "CONNECTION_TIMEOUT", fallback=60)
+            self._configuration["connection_timeout"] = int(parser.get(section, "CONNECTION_TIMEOUT", fallback=60))
```
Comment (pmrv) [inline pyiron_base/settings/generic.py]:
```suggestion
            self._configuration["connection_timeout"] = parser.getint(section, "CONNECTION_TIMEOUT", fallback=60)
```

File: pyiron_base/settings/generic.py [hunk 2]
```diff
@@ -292,12 +292,12 @@ def _config_parse_file(self, config_file):
...
         elif self._configuration["sql_type"] == "SQLalchemy":
             self._configuration["sql_connection_string"] = parser.get(
                 section, "CONNECTION"
             )
-            self._configuration["connection_timeout"] = parser.get(section, "CONNECTION_TIMEOUT", fallback=60)
+            self._configuration["connection_timeout"] = int(parser.get(section, "CONNECTION_TIMEOUT", fallback=60))
```
Comment (pmrv) [inline pyiron_base/settings/generic.py]:
```suggestion
            self._configuration["connection_timeout"] = parser.getint(section, "CONNECTION_TIMEOUT", fallback=60)
```

**Pull request merged**